resources/views/initiate-payment.blade.php

```diff
@@ -0,0 +1,22 @@
+<form method="POST" action="{{ route('payments') }}" accept-charset="UTF-8" class="form-horizontal d-none" role="form">
+
+
+    <input type="hidden" name="metadata" value="{{ json_encode($array = ['invoiceId' => $fee->id]) }}">
+
+
+    <input type="hidden" name="email" value="{{Auth::user()->email}}"> {{-- required --}}
+    <input type="hidden" name="orderID" value="345">
+
+
+    <input type="hidden" name="amount" value="{{$fee->total}}"> {{-- required in kobo --}}
+
+    <input type="hidden" name="currency" value="NGN">
+
+    <input type="hidden" name="reference" value="{{ Paystack::genTranxRef() }}">
+    {{ csrf_field() }}
+
+
+    <button class="btn btn-success btn-lg btn-block" type="submit" value="Pay Now!">
+
+        <i class="fa fa-plus-circle fa-lg"></i> Pay Now!</button>
+</form>
```

Comment:
I don't understand what this view is doing.

Doesn't make sense.

Why re you generating a ref again.

please follow the instruction in this link and use the popup method

https://paystack.com/docs/payments/accept-payments/#popup